### PR TITLE
Add MultipleFilePicker method & MPFile.toByteArray()

### PR DESCRIPTION
**Pull request by Shahriyar13**
tested on iOS, Android, macosX64 and JVM(macos, windows)

### CODE REVIEW

**Comment by Shahriyar13** [comment]
Web tested on Safari

**Comment by Shahriyar13** [comment]
JVM tested on mac

**Comment by StefanOltmann** [comment]
Will it work on Windows?

**Comment by Shahriyar13** [comment]
> Will it work on Windows?

Yes, but You can choose multiple files only if they are all located in the same folder.

**Comment by Shahriyar13** [comment]
@shalva97 @Wavesonics could you check my pull request and accept it please?

**Review by shalva97 — COMMENTED** [review]
Have not looked at it fully yet. Iam busy these days but I should have some time today and tomorrow

File: .run/iOS Example.run.xml [hunk 1]
```diff
@@ -0,0 +1,7 @@
+<component name="ProjectRunConfigurationManager">
```
[the remark this answers is not in the document]
Comment (Shahriyar13) [reply .run/iOS Example.run.xml:1]: When You running it in Xcode it's work, also you can skip it by clicking on Debug skip warning.

File: mpfilepicker/src/androidMain/kotlin/com/darkrockstudios/libraries/mpfilepicker/AndroidFilePicker.kt [hunk 2]
```diff
@@ -42,6 +45,43 @@ public actual fun FilePicker(
...
+public actual fun MultipleFilePicker(
+	show: Boolean,
+	initialDirectory: String?,
+	fileExtensions: List<String>,
+	onFileSelected: FilesSelected
+) {
+	val launcher = rememberLauncherForActivityResult(contract = ActivityResultContracts.OpenMultipleDocuments()) { result ->
+
+		val files = result.mapNotNull { uri ->
+			uri.path?.let {path ->
```
[the remark this answers is not in the document]
Comment (Shahriyar13) [reply mpfilepicker/src/androidMain/kotlin/com/darkrockstudios/libraries/mpfilepicker/AndroidFilePicker.kt:58]: URI.path is nullable, so if it's happened I will skip it by mapNotNull

File: examples/macosX64/src/macosX64Main/kotlin/main.kt [hunk 3]
```diff
@@ -40,9 +53,14 @@ fun main() {
...
-				FilePicker(show, fileExtensions = listOf("jpg", "png", "plist")) { file ->
-					pathChosen = file?.path ?: "none selected"
-					show = false
+				FilePicker(showSingleFile, fileExtensions = listOf("jpg", "png", "plist")) { file ->
+					singleFilePathChosen = file?.path ?: "none selected"
+					showSingleFile = false
+				}
+
+				MultipleFilePicker(showMultipleFile, fileExtensions = listOf("jpg", "png", "plist")) { files ->
+					multipleFilesPathsChosen = files?.map { it.path + "/n" } ?: listOf()
```
Comment (shalva97) [inline examples/macosX64/src/macosX64Main/kotlin/main.kt:62]: should be \ instead
```suggestion
					multipleFilesPathsChosen = files?.map { it.path + "\n" } ?: listOf()
```

File: examples/jvm/src/main/kotlin/Main.kt [hunk 4]
```diff
@@ -37,9 +51,14 @@ fun main() = application {
...
-	FilePicker(show, fileExtensions = listOf("jpg", "png")) { file ->
-		pathChosen = file?.path ?: "none selected"
-		show = false
+	FilePicker(showSingleFile, fileExtensions = listOf("jpg", "png")) { file ->
+		pathSingleChosen = file?.path ?: "none selected"
+		showSingleFile = false
+	}
+
+	MultipleFilePicker(showMultiFile, fileExtensions = listOf("jpg", "png")) { files ->
+		pathMultiChosen = files?.map { it.path + "/n" } ?: emptyList()
```
Comment (shalva97) [inline examples/jvm/src/main/kotlin/Main.kt:60]:
```suggestion
		pathMultiChosen = files?.map { it.path + "\n" } ?: emptyList()
```

File: examples/ios/src/iosMain/kotlin/main.ios.kt [hunk 5]
```diff
@@ -52,6 +71,21 @@ fun MainViewController(): UIViewController = ComposeUIViewController {
 
 			/////////////////////////////////////////////////////////////////
 
+			var nonComposeMultipleFileChosen by remember { mutableStateOf(listOf("")) }
+
+			Button(onClick = {
+				MainScope().launch {
+					nonComposeMultipleFileChosen = launchFilePicker(fileExtensions = fileType, allowMultiple = true)
+						.map { it.path + "/n" }
```
Comment (shalva97) [inline examples/ios/src/iosMain/kotlin/main.ios.kt:79]:
```suggestion
						.map { it.path + "\n" }
```

**Review by shalva97 — CHANGES_REQUESTED** [review]

**Comment by shalva97** [comment]
It's great. Tested it on MacOS and works perfectly.

Was there a specific reason that you have added a new function `MultipleFilePicker` instead of adding a new argument to an old `FilePicker` function? At least this is how I would do it...

Don't pay attention to CI/CD. I have canceled it because for some reason build takes too much time.

Accept few code suggestions and I will merge it.

**Comment by shalva97** [comment]
yes, that makes sense.

@Wavesonics Should I merge this directly? for me it looks good

**Pull request merged**